components/raftstore/src/store/fsm/apply.rs

```diff
@@ -547,9 +554,18 @@ where
         self.host
             .on_flush_applied_cmd_batch(batch_max_level, cmd_batch, &self.engine);
         // Invoke callbacks
+        let now = Instant::now();
         for (cb, resp) in cb_batch.drain(..) {
-            cb.invoke_with_response(resp)
+            if let Some(times) = cb.get_request_times() {
```

Comment:
Shouldn't that be controlled by waterfall metrics?

Reply:
No. The `store_time` and `apply_time` metrics are open by default. The sum of their avg value should be equal to `async_write_time`.